libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -6,398 +6,135 @@ import { ValidationMetadata } from 'class-validator/metadata/ValidationMetadata'
-          }
-        });
-        if (formGroupField[0] instanceof DynamicFormGroup) {
-          formGroupFields[key] = formGroupField[0];
-        } else {
-          if (formGroupField[0] instanceof FormArray) {
-            formGroupFields[key] = formGroupField[0];
-          } else {
-            formGroupFields[key] = [formGroupField[0], formGroupField.filter((item, index) => index !== 0)];
-          }
-        }
-      });
-    return formGroupFields;
+
+  // Getters & Setters
+  set externalErrors(externalErrors: ShortValidationErrors) {
+    this._externalErrors = externalErrors;
+    this.validate();
+  }
+  get externalErrors(): ShortValidationErrors {
+    return this._externalErrors;
   }
 
-  private onlyFields(fields: { [key: string]: any }) {
-    const newFields = {};
-    if (fields !== undefined) {
-      Object.keys(fields).forEach(key => {
-        if (fields[key] instanceof DynamicFormGroup) {
-          newFields[key] = this.onlyFields((fields[key] as DynamicFormGroup<any>).fields);
-        } else {
-          if (fields[key] instanceof FormArray) {
-            if ((fields[key] as FormArray).controls[0] instanceof DynamicFormGroup) {
-              newFields[key] = this.onlyFields(((fields[key] as FormArray).controls[0] as DynamicFormGroup<any>).fields);
-            } else {
-              newFields[key] = (fields[key] as FormArray).controls[0].value;
-            }
-          } else {
-            newFields[key] = fields[key];
-          }
-        }
-      });
-    }
-    return newFields;
+  set validatorOptions(validatorOptions: ValidatorOptions) {
+    this._validatorOptions = validatorOptions;
+    this.validate();
   }
-  get object() {
-    const object = this._object ? this.classToClass(this._object) : new this.factoryModel();
-    if (object !== undefined) {
-      Object.keys(this.controls).forEach(key => {
-        if (this.controls[key] instanceof DynamicFormGroup) {
-          object[key] = (this.controls[key] as DynamicFormGroup<any>).object;
-        } else {
-          if (this.controls[key] instanceof FormArray) {
-            object[key] = [];
-            for (let i = 0; i < (this.controls[key] as FormArray).controls.length; i++) {
-              if ((this.controls[key] as FormArray).controls[i] instanceof DynamicFormGroup) {
-                object[key].push(
-                  ((this.controls[key] as FormArray).controls[i] as DynamicFormGroup<any>).object
-                );
-              } else {
-                object[key].push(
-                  (this.controls[key] as FormArray).controls[i].value
-                );
-              }
-            }
-          } else {
-            object[key] = this.controls[key].value;
-          }
-        }
-      });
-    }
-    return this.plainToClass(this.factoryModel, object);
+  get validatorOptions(): ValidatorOptions {
+    return this._validatorOptions;
   }
+
   set object(object: TModel) {
-    if (object instanceof this.factoryModel) {
```

Comment:
Moved much of this to a private function at the end of the file to slim down the get/set statements at the top where the public API is defined